nodestream/project/pipeline_scope.py

```diff
@@ -26,26 +26,34 @@ def __init__(
         pipelines: List[PipelineDefinition],
         persist: bool = True,
         config: ScopeConfig = None,
-        targets: List[str] = None,
+        targets: Set[str] = frozenset(),
     ) -> None:
         self.persist = persist
         self.name = name
         self.config = config
         self.targets = targets
         self.pipelines_by_name: Dict[str, PipelineDefinition] = {}
         for pipeline in pipelines:
+            if self.targets is not None:
+                pipeline.targets = set(pipeline.targets | self.targets)
```

Comment:
Minor thing here, but I think `set | set` should produce a `set` making calling `set` redundant.